File: src/main/java/com/nerdearla/workshop/payment/mapper/PaymentResponseMapper.java
package com.nerdearla.workshop.payment.mapper;

import com.nerdearla.workshop.payment.model.PaymentResponse;
import com.nerdearla.workshop.payment.model.Payment;
import org.springframework.stereotype.Component;

@Component
public class PaymentResponseMapper {

    public PaymentResponse map(Payment payment) {
        return new PaymentResponse(payment.getPaymentId());
    }
}

File: src/main/java/com/nerdearla/workshop/payment/mapper/PaymentResponseMapper.kt
package com.nerdearla.workshop.payment.mapper

import com.nerdearla.workshop.payment.model.Payment
import com.nerdearla.workshop.payment.model.PaymentResponse
import org.springframework.stereotype.Component

@Component
class PaymentResponseMapper {
    fun map(payment: Payment): PaymentResponse =
        with(payment) {
            PaymentResponse(
                paymentId = id
            )
        }

}
